### PR TITLE
Avoid spurious libgcc-s1 Debian package dependency

### LINKED ISSUE #5875: Ubuntu 18.04 deb depends on libgcc-s1 which isn't in 18.04, it should depend on libgcc1 instead
CBMC version: 5.24 (also older packages)
Operating system: Ubuntu 18.04
Exact command line resulting in the issue: dpkg -i ubuntu-18.04-cbmc-5.24.0-Linux.deb
What behaviour did you expect: CBMC gets installed
What happened instead: dpkg returns: cbmc : Depends: libgcc-s1 (>= 3.4) but it is not installable

libgcc-s1 isn't in Ubuntu 18.04: https://packages.ubuntu.com/search?suite=xenial&arch=amd64&keywords=libgcc-s1
Update: Sorry, Ubuntu 18.04 is bionic, not xenial: https://packages.ubuntu.com/search?suite=bionic&arch=amd64&keywords=libgcc-s1

The file libgcc_s.so.1 does exist, but is provided by the libgcc1 library in 18.04.

I tried manually extracting and running cbmc, and it does seem to work, so `dpkg -i --force-depends` should work.

### PR DESCRIPTION
**Pull request by tautschnig**
Marked as Draft (and do-not-merge, do-not-review), includes extra debug output.

Do not add a strict dependency on gcc, which in a GitHub container isn't
necessarily the same version as in a pristine Ubuntu system.

Fixes: #5875

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: .github/workflows/pull-request-checks.yaml [hunk 1]
```diff
@@ -375,6 +375,7 @@ jobs:
         run: | 
           sudo apt-get update
           sudo apt-get install --no-install-recommends -y g++ flex bison cmake ninja-build maven jq libxml2-utils dpkg-dev ccache
+          sudo apt-get --purge remove -s libgcc-s1
```
Comment (hannes-steffenhagen-diffblue) [inline .github/workflows/pull-request-checks.yaml]: Huh, that seems screwy? So the github actions ubuntu-18.04 container isn't necessarily actually ubuntu-18.04?

**Comment by hannes-steffenhagen-diffblue** [comment]
For curiosities sake, how did you detemine this was the issue?

**Comment by kroening** [comment]
The cleaner answer here could be do build that package in a docker container (Github actions does this). The problem with the proposed tweak is that Github may well choose to make further changes to their standard image, and then the issue is back.

**Comment by hannes-steffenhagen-diffblue** [comment]
@tautschnig Following @kroening's comment here, I believe the correct thing to do would be to specify this running on the ubuntu:18.04 docker image via the [container](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer) attribute, then we don't have to remove packages which seems slightly sketchy anyway.

TBH I didn't realise that the actions images had so much nonstandard stuff in them when we came up with these actions.

**Comment by tautschnig** [comment]
> @tautschnig Following @kroening's comment here, I believe the correct thing to do would be to specify this running on the ubuntu:18.04 docker image via the [container](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer) attribute, then we don't have to remove packages which seems slightly sketchy anyway.
[...]

I tried this avenue, but then our use of submodules requires git 2.18, which isn't in Ubuntu 18.04. Thus sticking with the force-downgrade option for now. Marking the PR as ready-for-review as the logs confirmed that libgcc-s1 no longer is among the dependencies being generated.

**Review by NlightNFotis — APPROVED** [review]
LGTM

**Comment by thomasspriggs** [comment]
> > @tautschnig Following @kroening's comment here, I believe the correct thing to do would be to specify this running on the ubuntu:18.04 docker image via the [container](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer) attribute, then we don't have to remove packages which seems slightly sketchy anyway.
> > [...]
> 
> I tried this avenue, but then our use of submodules requires git 2.18, which isn't in Ubuntu 18.04. Thus sticking with the force-downgrade option for now. Marking the PR as ready-for-review as the logs confirmed that libgcc-s1 no longer is among the dependencies being generated.

Can you further explain what aspects of submodules aren't working with git 2.17? I am using ubuntu 18.04 locally and I haven't seen any submodule related issues.

**Comment by tautschnig** [comment]
> > > @tautschnig Following @kroening's comment here, I believe the correct thing to do would be to specify this running on the ubuntu:18.04 docker image via the [container](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer) attribute, then we don't have to remove packages which seems slightly sketchy anyway.
> > > [...]
> > 
> > 
> > I tried this avenue, but then our use of submodules requires git 2.18, which isn't in Ubuntu 18.04. Thus sticking with the force-downgrade option for now. Marking the PR as ready-for-review as the logs confirmed that libgcc-s1 no longer is among the dependencies being generated.
> 
> Can you further explain what aspects of submodules aren't working with git 2.17? I am using ubuntu 18.04 locally and I haven't seen any submodule related issues.

See https://github.com/diffblue/cbmc/runs/2021105147 for one of the past logs. That says:
```
The repository will be downloaded using the GitHub REST API
To create a local Git repository instead, add Git 2.18 or higher to the PATH
Error: Input 'submodules' not supported when falling back to download using the GitHub REST API. To create a local Git repository instead, add Git 2.18 or higher to the PATH.
```

**Comment by hannes-steffenhagen-diffblue** [comment]
@tautschnig That seems like an issue with the github action we use for fetching the code. I don't think that's a real blocker for building on 18.04 proper, worst case we can just work around that.

**Comment by tautschnig** [comment]
> @tautschnig That seems like an issue with the github action we use for fetching the code. I don't think that's a real blocker for building on 18.04 proper, worst case we can just work around that.

Up to you. Can I suggest we put a fix in place before merging #5883 today? This might be merging this PR and doing a cleaner solution later on, or someone making the container solution work today.

**Comment by hannes-steffenhagen-diffblue** [comment]
@tautschnig seems good to me, plan is to merge this and put building in docker in the mid-term

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]

**Pull request merged**